### ISSUE TITLE
[Barcode] Invalid barcode rawValue

### ISSUE THREAD

**Issue by JrmyDev** [bharat-biradar/Google-Ml-Kit-plugin#100]
Hi,
My customer have issue on iPhone 6S while scanning barcode. 
Sometimes, it returns barcode rawValue like `{"typeNumber": "0", "errorCode": "3", "version": "229"}` 
`version` sometimes change.
I think errorCode may be related to MLkitException.ErrorCode
https://developers.google.com/android/reference/com/google/mlkit/common/MlKitException?hl=fr

Flutter version 2.2.3
Google ml kit version 0.6.0

I think on some condition, barcode detector from google ml kit Barcode Scanner could return error on detection.

- `com.b.biradar.google_ml_kit.vision.BarcodeDetector.java` l.213 for Android
- `BarcodeScanner.m` l.40 for iOS

A fix might be apply by catching error instead of returning it as a barcode in `barcode_scanner.dart` like that :
```
Future<List<Barcode>> processImage(InputImage inputImage) async {
    _isOpened = true;
    var result;
    try {
       result = await Vision.channel.invokeMethod(
          'vision#startBarcodeScanner', <String, dynamic>{
        'formats': barcodeFormats,
        'imageData': inputImage._getImageData()
      });
    } on PlatformException catch(e) {
        print(e);
        throw e;
    }

    final barcodesList = <Barcode>[];
    for (dynamic item in result) {
      barcodesList.add(Barcode._fromMap(item));
    }

    return barcodesList;
  }
```

I'm not sure about it, because I can't reproduce this problem.

**Comment by fbernaly** [comment]
I have the same iPhone, could you share the barcodes your client is scanning so I can try to reproduce.

**Comment by fbernaly** [comment]
@JrmyDev: We just released 0.7.0 we a few fixes for barcode, could you give it a try? also, share your barcode so I can test on my iPhone 6S and try to reproduce

**Comment by JrmyDev** [comment]
@fbernaly For now, I'm not able to share you the barcode because I retrieve only the error remotely.
I will push a patch to retrieve more information and send you more information asap

**Comment by JrmyDev** [comment]
I was able to reproduce it on Android.
When I instanciate my barcode scanner without provinding barcode format :
`BarcodeScanner barcodeScanner = GoogleMlKit.vision.barcodeScanner();`

The scanner detect some object or shape as a barcode unknown or text type and return this kind of error
`"typeNumber": "0", "errorCode": "3", "version": "229"}`

It seems that I resolve it by submitting a barcode format :
`BarcodeScanner barcodeScanner = GoogleMlKit.vision.barcodeScanner([BarcodeFormat.code128])`

**Comment by jbx1** [comment]
This problem seems to be more prevalent on 0.7.0 than on 0.6.0 for some reason.

**Comment by jbx1** [comment]
> @JrmyDev: We just released 0.7.0 we a few fixes for barcode, could you give it a try? also, share your barcode so I can test on my iPhone 6S and try to reproduce

Actually 0.7.0 makes it worse. I didn't have this issue, and after upgrading from 0.6.0 to 0.7.0 I started to face this issue.

**Comment by JrmyDev** [comment]
> > @JrmyDev: We just released 0.7.0 we a few fixes for barcode, could you give it a try? also, share your barcode so I can test on my iPhone 6S and try to reproduce
> 
> Actually 0.7.0 makes it worse. I didn't have this issue, and after upgrading from 0.6.0 to 0.7.0 I started to face this issue.

Try to define the list of barcode type to handle like 
`BarcodeScanner barcodeScanner = GoogleMlKit.vision.barcodeScanner([BarcodeFormat.code128])`

**Comment by jbx1** [comment]
> > > @JrmyDev: We just released 0.7.0 we a few fixes for barcode, could you give it a try? also, share your barcode so I can test on my iPhone 6S and try to reproduce
> > 
> > 
> > Actually 0.7.0 makes it worse. I didn't have this issue, and after upgrading from 0.6.0 to 0.7.0 I started to face this issue.
> 
> Try to define the list of barcode type to handle like
> `BarcodeScanner barcodeScanner = GoogleMlKit.vision.barcodeScanner([BarcodeFormat.code128])`

Yeah did that, and also added a check that if the value starts with `{"typeNumber"` ignore it and continue scanning as a workaround.

**Comment by fbernaly** [comment]
We released a new version of [google_ml_kit](https://github.com/bharat-biradar/Google-Ml-Kit-plugin/tree/master/packages/google_ml_kit) in which we have split all the plugins in multiple plugins to avoid removing the models for API you are not using, go to our readme and start using the new plugin you need.

Start using:
https://github.com/bharat-biradar/Google-Ml-Kit-plugin/tree/master/packages/google_mlkit_barcode_scanning

More details here:

https://github.com/bharat-biradar/Google-Ml-Kit-plugin/tree/master/packages/google_ml_kit
